Prevent folder uploads via drag-and-drop

Randomlyclueless opened the pull request:
This PR improves the user experience when users drag and drop a folder into the
compose box.

Previously, dropping a folder could result in unclear or inconsistent behavior,
such as an “unknown error” banner or partial upload attempts depending on the
browser. This change explicitly detects directory drops at the drag-and-drop
handler level and rejects them early with a clear, user-facing error message.

The new banner text is:

> **A folder can't be uploaded. Instead, please upload the files you need.**

This ensures users immediately understand what went wrong and how to proceed.

Fixes #22564.

## How changes were tested

- Ran the Zulip development server locally
- Dragged and dropped a folder into:
  - the compose box
  - the message edit box
  - the main app area
- Verified that folder uploads are rejected with the expected error banner
- Verified that normal file drag-and-drop behavior remains unchanged

### Checklist

- [x] Tests written or not needed
- [x] Manual testing performed
- [x] Screenshots included for UI changes
- [x] Commit message follows Zulip guidelines

## Screenshot

<img width="1905" height="948" alt="zulipupload" src="https://github.com/user-attachments/assets/ce5b0dfa-c51a-4f6e-8ffb-6e0cab77578b" />

alya commented:
Please clean up your commit history and post again to request a review. See [here](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html) for guidelines.

You will also need to complete the self-review checklist.

Randomlyclueless commented:
Hi @alya, thanks for the feedback — and sorry for the earlier misses.
I’ve now updated the commit message to fully follow Zulip’s guidelines (including the issue reference), corrected the PR checklist to match the template, and replaced the screenshot so it reflects the final behavior.
The PR should now address all the points raised and be ready for another review. Thanks!
